docs/admin/memory.rst

```diff
@@ -40,6 +47,10 @@ Consider the following when determining the right value:
   collection times **increase** with a bigger heap, leading to **increased**
   latency. Therefore, your heap size shouldn't be too large.
 
+- The optimal heap size is workload dependent. Be sure to test carefully
+  if you want to change the default, as wrong values may cause CrateDB to crash
+  with OutOfMemory errors.
+
```

Comment:
Isn't this just the same point as below?

> Therefore, the heap region must be large enough to hold the biggest
  result sets in memory. The size of the intermediate and final result sets
  depend entirely on the type of queries you are running.

Is the same as "optimal heap size is workload dependent" but with more details.

The "wrong values may cause CrateDB to crash with OutOfMemory errors." also
doesn't exactly raise confidence in the product - or give the impression that
we know what we're doing.

We should give concrete min values for both heap and off-heap, and after that
users should receive circuit breaker errors and not have their nodes crash.